rest-service/manager_rest/upload_manager.py

```diff
@@ -659,9 +660,26 @@ def _prepare_and_process_doc(self,
                     '{version}'.format(archive_name=new_plugin.archive_name,
                                        package_name=new_plugin.package_name,
                                        version=new_plugin.package_version))
+            if self._is_plugin_installing(new_plugin, plugin):
+                raise manager_exceptions.ConflictError(
+                    'a plugin archive by the name of {archive_name} for '
+                    'package with name {package_name} and version {version} '
+                    'is currently being installed'.format(
+                        archive_name=new_plugin.archive_name,
+                        package_name=new_plugin.package_name,
+                        version=new_plugin.package_version))
+        new_plugin.archive_name = INSTALLING_PREFIX + new_plugin.archive_name
```

Comment:
Maybe use '{0}{1}'.format instead of `+` as it is more consistent to the rest of our code?
We don't usually use `+`

Reply:
using the .format approach is better memory wise

Reply:
not always, in this case `+` uses less memory than `format` because `+` involves creating 3 strings: the 2 arguments, and the resulting output
and `format` involves creating 4 strings: the 2 arguments, the resulting output, and the `'{0}{1}'` format string :P

that doesn't really matter though of course, and str.format is great and fits here